### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by Nadoshka**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -30,18 +32,116 @@
...
+            text: contributor.login,
+            class: 'contributor-name',
+          });
+          const contributionsNum = createAndAppend('td', row);
+          createAndAppend('span', contributionsNum, {
+            text: contributor.contributions,
+            class: 'contributor-number',
+          });
+        })
+        .catch(err => {
```
Comment (remarcmij) [inline homework/index.js]: This line produces an error message in the browser console:

```
Uncaught (in promise) TypeError: Cannot read property 'catch' of undefined
    at index.js:97
```

File: homework/style.css [hunk 2]
```diff
@@ -1,3 +1,86 @@
 .alert-error {
```
Comment (remarcmij) [inline homework/style.css]: You have still some work to do to make your app responsive:

image

File: homework/index.js [hunk 3]
```diff
@@ -30,18 +32,116 @@
...
+    });
+  }
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+    fetchJSON(url)
+      .then(data => {
```
Comment (remarcmij) [inline homework/index.js]: Renaming `data` to `repositories` will make your code more readable.

File: homework/index.js [hunk 4]
```diff
@@ -30,18 +32,116 @@
...
+    });
+    createAndAppend('li', listRepoInfo, { text: `Forks: ${selectedRepo.forks}`, class: 'li' });
+    createAndAppend('li', listRepoInfo, {
+      text: `Updated: ${selectedRepo.updated_at}`,
+      class: 'li',
+    });
+  }
+
+  // create (right box) as a table to render contributor Info
+  function createContributor(selectedRepo, repoContainer, root) {
```
Comment (remarcmij) [inline homework/index.js]: You are handling **_all_** contributors here -> `createContributors` (mind the plural 's').

File: homework/index.js [hunk 5]
```diff
@@ -30,18 +32,116 @@
...
+          const repoContainer = document.getElementsByClassName('repo-container')[0];
+          removeChildren(repoContainer);
+
+          // getting the selected repo using the value of event.target which is select element
+          const selectedRepo = data[event.target.value];
+
+          // rendering info of repository which we chose
+          createRepoInfo(selectedRepo, repoContainer);
+          createContributor(selectedRepo, repoContainer, root);
+        });
```
Comment (remarcmij) [inline homework/index.js]: You could move all of this code (lines 120-131) just before line 139. If you do that you don't need line 122 to get a reference to the `repoContainer` as you have it already from line 133.

File: homework/index.js (hunk 1 above)
Comment (Nadoshka) [reply homework/index.js]: Thanks, U are right! It's done!

File: homework/style.css (hunk 2 above)
Comment (Nadoshka) [reply homework/style.css]: Thanks! I will fix it soon!

File: homework/index.js (hunk 3 above)
Comment (Nadoshka) [reply homework/index.js]: Thanks! I agree! It's done!

File: homework/index.js (hunk 5 above)
Comment (Nadoshka) [reply homework/index.js]: Thanks a lot! that is handy. It's done! :)

File: homework/index.js (hunk 4 above)
Comment (Nadoshka) [reply homework/index.js]: :

**Pull request closed without merging**